cmake/test_util.cmake

```diff
@@ -43,7 +43,7 @@ function(create_standard_test_cxx)
   add_executable(test_${TEST_NAME} ${TEST_SOURCES})
   target_link_libraries(test_${TEST_NAME} PUBLIC musica::musica GTest::gtest_main)
   if(MUSICA_ENABLE_OPENMP)
-    target_link_libraries(test_${TEST_NAME} PUBLIC OpenMP::OpenMP_CXX OpenMP::OpenMP_Fortran)
+    target_link_libraries(test_${TEST_NAME} PUBLIC OpenMP::OpenMP_CXX)
   endif()
```

Comment:
Hi @mattldawson and @K20shores, what do you think about this change? Matt and I briefly discussed about this. The issue is that `OpenMP::OpenMP_Fortran` is not found when `test_micm_c_api` is built. I removed it because this line belongs to `function(create_standard_test_cxx)` and I assumed that it shouldn't need `OpenMp_Fortran` for c++ tests. 
Matt mentioned we could specify fortran as one of the languages in the top-level `CMakeLists.txt` and update `find_package` in `cmake/dependencies.cmake`
```
project(musica-distribution VERSION 0.7.0 LANGUAGES C CXX Fortran)
find_package(OpenMP REQUIRED COMPONENTS C CXX Fortran)
```
I think that makes sense although I think it is useful to separate the fortran stuff from `MUSICA-C` as it is since we eventually pursue fortran-free. I'd like to hear what you all think.

Reply:
I would like to keep fortran separated, but with the inclusion of tuvx into the musica-c library that line is blurred. Because of that I think we should go with Matt's suggestion. The other option is to check if TUVX is enabled before we check for OpenMP. If it is, enable the fortran language so that the fortran part of OpenMP is found, if that's possible

Reply:
I created [a separate issue](https://github.com/NCAR/musica/issues/179) to tackle this.